Library/Homebrew/linkage_checker.rb

```diff
@@ -1,42 +1,73 @@
 require "keg"
 require "formula"
+require "linkage_cache_store"
 
 class LinkageChecker
-  attr_reader :undeclared_deps
-
-  def initialize(keg, formula = nil)
+  def initialize(keg, db, use_cache = false, formula = nil)
     @keg = keg
     @formula = formula || resolve_formula(keg)
-    @brewed_dylibs = Hash.new { |h, k| h[k] = Set.new }
-    @system_dylibs = Set.new
-    @broken_dylibs = []
-    @broken_deps = Hash.new { |h, k| h[k] = [] }
-    @variable_dylibs = Set.new
-    @indirect_deps = []
-    @undeclared_deps = []
-    @reverse_links = Hash.new { |h, k| h[k] = Set.new }
-    @unnecessary_deps = []
-    check_dylibs
+    @store = LinkageStore.new(keg.name, db)
+    flush_cache_and_check_dylibs unless use_cache
+  end
+
+  # 'Hash-type' cache values
+
+  def brewed_dylibs
```

Comment:
Can you make all of these methods `private` that were before? Trying to reduce the `public` surface of this class since you first opened the PR. Thanks!